swift/ql/lib/codeql/swift/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -41,6 +43,21 @@ private class ExprNodeImpl extends ExprNode, NodeImpl {
   override string toStringImpl() { result = expr.toString() }
 
   override DataFlowCallable getEnclosingCallable() { result = TDataFlowFunc(n.getScope()) }
+
+  override DataFlowType getTypeImpl() {
+    result = getDataFlowType(expr.getType())
+    or
+    result.asCallable().asSourceCallable() = expr
+    or
+    result.asCallable().asSourceCallable() = expr.(DeclRefExpr).getDecl()
```

Comment:
Why do we need both cases here? Is that to handle each callable having their own type?

In any case, a comment on this would be helpful